sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeClientOptions.cs

```diff
@@ -1,19 +1,222 @@
     {
+        /// <summary>
+        /// The Latest service version supported by this client library.
+        /// </summary>
+        internal const ServiceVersion LatestVersion = ServiceVersion.V2019_02_02;
 
+        /// <summary>
+        /// The versions of Azure Data Lake Sevice supported by this client
+        /// library.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services" />.
+        /// </summary>
+        public enum ServiceVersion
+        {
+#pragma warning disable CA1707 // Identifiers should not contain underscores
+            /// <summary>
+            /// The 2019-02-02 service version described at
+            /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services#version-2019-02-02" />
+            /// </summary>
+            V2019_02_02 = 1
+#pragma warning restore CA1707 // Identifiers should not contain underscores
+        }
+
+        /// <summary>
+        /// Gets the <see cref="ServiceVersion"/> of the service API used when
+        /// making requests.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services" />.
+        /// </summary>
+        public ServiceVersion Version { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataLakeClientOptions"/>
+        /// class.
+        /// </summary>
+        /// <param name="version">
+        /// The <see cref="ServiceVersion"/> of the service API used when
+        /// making requests.
+        /// </param>
+        public DataLakeClientOptions(ServiceVersion version = LatestVersion)
+        {
+            Version = version == ServiceVersion.V2019_02_02 ? version : throw Errors.VersionNotSupported(nameof(version));
+            this.Initialize();
+            AddHeadersAndQueryParameters();
+        }
+
+        /// <summary>
+        /// Gets or sets the secondary storage <see cref="Uri"/> that can be read from for the storage account if the
+        /// account is enabled for RA-GRS.
+        ///
+        /// If this property is set, the secondary Uri will be used for GET or HEAD requests during retries.
+        /// If the status of the response from the secondary Uri is a 404, then subsequent retries for
+        /// the request will not use the secondary Uri again, as this indicates that the resource
+        /// may not have propagated there yet. Otherwise, subsequent retries will alternate back and forth
+        /// between primary and secondary Uri.
+        /// </summary>
+        public Uri GeoRedundantSecondaryUri { get; set; }
+
+        /// <summary>
+        /// Add headers and query parameters in <see cref="DiagnosticsOptions.LoggedHeaderNames"/> and <see cref="DiagnosticsOptions.LoggedQueryParameters"/>
+        /// </summary>
+        private void AddHeadersAndQueryParameters()
```

Comment:
Do we need to add any new headers or query params for DataLake?

Reply:
See my comment in response to Josh below, there are headers that are just used by Data Lake.

Reply:
Wait, do I also need to add the Data Lake-specific query parameters, too?

Reply:
Yes, it's an allow-list.  Anything we don't provide here will be REDACTED in logs.

Reply:
Ok, I went back and added the Data Lake-specific query parameters.